collab-service/src/message_handler/internal/internal_message_types.d.ts

```diff
@@ -0,0 +1,25 @@
+const enum ConnectionFlag {
+  NONE,
+  JOIN,
+  ACK,
+}
+
+declare type TunnelMessage = {
+  sender: string,
+  nick: string,
+  data: Uint8Array,
+  flag: ConnectionFlag,
+};
+
+declare type TunnelInternalMessage = {
+  sender: string,
+  nick: string,
```

Comment:
Likewise, I feel that there is no need to introduce the additional `nick` field.